infra/azure/modules/batch/main.tf

```diff
@@ -39,6 +39,23 @@ resource "azurerm_shared_image" "batch_worker" {
   }
 }
 
+resource "azurerm_shared_image" "batch_worker_22_04" {
+  name                = "batch-worker-22-04"
+  gallery_name        = azurerm_shared_image_gallery.batch.name
+  resource_group_name = var.resource_group.name
+  location            = var.resource_group.location
+  os_type             = "Linux"
+  specialized         = false
+
+  hyper_v_generation = "V2"
+
+  identifier {
+    publisher = "Hail"
+    offer     = "BatchWorker"
+    sku       = "Ubuntu-22.04-LTS"
+  }
+}
```

Comment:
I think we had a misunderstanding of what the shared image definition is doing. An image definition does not tie itself to a particular base VM image like Ubuntu 20.04 or Ubuntu 22.04, it just defines a [logical grouping of concrete image versions](https://learn.microsoft.com/en-us/azure/virtual-machines/shared-image-galleries?tabs=azure-cli#image-definitions) with some broad characteristics defined like whether the image versions are Linux or Windows and which hypervisor generation is needed to run them.

This `identifier` block allows us to define a canonical name for the image definition that we are creating, and AFAICT is not intended to reference an existing marketplace identifier. So the existing definition doesn't actually give us a Ubuntu 20.04 base VM image.

Where the base VM image is actually defined is where we create the VM in `az-create-worker-image.sh` with `az vm create` (this makes sense in hindsight because we don't actually reference the image definition in the `az vm create` call so how could it know to pick 20.04 vs 22.04?). In that call we have been using `--image UbuntuTLS`. I think this is Ubuntu 18.04, because that's what I get when running `lsb_release -a` on a production Azure VM. So in this PR I've changed that to be `Ubuntu2204` which seems to have done the trick. I think we still need a different image definition here though because Ubuntu 18.04 and 22.04 need different `hyper_v_generation`. Trying to save an Ubuntu 22.04 VM under the existing definition resulted in an error saying that the VM was incompatible with a V1 image definition.